runtime/compilers/rillv1beta/connector.go

```diff
@@ -72,13 +73,18 @@ func ExtractConnectors(ctx context.Context, projectPath string) ([]*Connector, e
 		// this can fail under cases such as full or host/bucket of URI is a variable
 		access, _ := connector.HasAnonymousAccess(ctx, &connectors.Env{}, src)
 		c := key{Name: src.Connector, Type: src.Connector, AnonymousAccess: access}
-		connectorMap[c] = true
+		uris, ok := connectorMap[c]
+		if !ok {
+			uris = make([]string, 0)
+		}
+		uris = append(uris, src.Properties["path"].(string))
```

Comment:
For current set of connectors where we need access, this is always guaranteed to exist. 
Based on your other comments, I have following questions:
1. When we introduce other connectors like `BigQuery` will we always show name of the source and not the URI for all connectors or will it be based on connector type ?
2. If this is based on the connector type, this needs to be driven from the `spec`. Any recommendation on what field to add in the spec ?

Reply:
Let's keep it simple for now and just show the source name. If the requirement arises in the future, we could have the connector/spec provide a way to format a "display name" for a source.

Reply:
Just discovered a corner case for embedded sources. Name will be an internal guid.

- May be we show URIs in all cases for now and revisit this if we add connectors in future where URI holds no value ?  or
- Show URIs for embedded sources and names on normal ones. Will be ugly but I don't think anybody will be using both kind of sources.

Reply:
Option 1 does not seem good. Two other options:

- Output URI if set, else source name
- Add the display name formatter and build logic there

Reply:
> Output URI if set, else source name

This one seems good to me. The fields can't even be derived from the connector `spec` now because of the embedded source use case. I can picture the formatter to have a similar logic as well.